Samples and tools fail to build with libc++ due to std::ignore

Simplified example:
```c++
$ cat a.cc
#include <tuple>

typedef struct {
  int foo : 8;
} test_t;

int main()
{
  test_t t;
  std::ignore = t.foo;

  return 0;
}

$ clang++ a.cc
a.cc:10:17: error: non-const reference cannot bind to bit-field 'foo'
  std::ignore = t.foo;
                ^~~~~
a.cc:4:7: note: bit-field is declared here
  int foo : 8;
      ^
/usr/include/c++/v1/tuple:1091:38: note: passing argument to parameter here
    const __ignore_t& operator=(_Tp&&) const {return *this;}
                                     ^
1 error generated.
```


Clang 9 reports:
```c++
samples/sample_hevc_fei_abr/src/dso/hevc2_ssdata.cpp:561:5: error:
      non-const reference cannot bind to bit-field 'log2CbSize'
    BS2_TRACE(cu.log2CbSize, cu.log2CbSize);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
samples/sample_hevc_fei_abr/include/dso/bs_reader2.h:187:44: note:
      expanded from macro 'BS2_TRACE'
#define BS2_TRACE(val, var) {std::ignore = (val);}
                                           ^~~~~
samples/sample_hevc_fei_abr/include/dso/hevc2_struct.h:832:11: note:
      bit-field is declared here
    Bs32u log2CbSize             : 8;
          ^
/usr/include/c++/v1/tuple:1091:38: note: passing argument to parameter here
    const __ignore_t& operator=(_Tp&&) const {return *this;}
                                     ^
samples/sample_hevc_fei_abr/src/dso/hevc2_ssdata.cpp:635:9: error:
      non-const reference cannot bind to bit-field 'inter_pred_idc'
        BS2_TRACE(pu.inter_pred_idc, pu.inter_pred_idc);
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
samples/sample_hevc_fei_abr/include/dso/bs_reader2.h:187:44: note:
      expanded from macro 'BS2_TRACE'
#define BS2_TRACE(val, var) {std::ignore = (val);}
                                           ^~~~~
samples/sample_hevc_fei_abr/include/dso/hevc2_struct.h:789:11: note:
      bit-field is declared here
    Bs16u inter_pred_idc : 2;
          ^
/usr/include/c++/v1/tuple:1091:38: note: passing argument to parameter here
    const __ignore_t& operator=(_Tp&&) const {return *this;}
                                     ^
samples/sample_hevc_fei_abr/src/dso/hevc2_ssdata.cpp:641:9: error:
      non-const reference cannot bind to bit-field 'w'
        BS2_TRACE(pu.w, pu.w);
        ^~~~~~~~~~~~~~~~~~~~~
samples/sample_hevc_fei_abr/include/dso/bs_reader2.h:187:44: note:
      expanded from macro 'BS2_TRACE'
#define BS2_TRACE(val, var) {std::ignore = (val);}
                                           ^~~~~
samples/sample_hevc_fei_abr/include/dso/hevc2_struct.h:784:11: note:
      bit-field is declared here
    Bs16u w : 8;
          ^
/usr/include/c++/v1/tuple:1091:38: note: passing argument to parameter here
    const __ignore_t& operator=(_Tp&&) const {return *this;}
                                     ^
samples/sample_hevc_fei_abr/src/dso/hevc2_ssdata.cpp:642:9: error:
      non-const reference cannot bind to bit-field 'h'
        BS2_TRACE(pu.h, pu.h);
        ^~~~~~~~~~~~~~~~~~~~~
samples/sample_hevc_fei_abr/include/dso/bs_reader2.h:187:44: note:
      expanded from macro 'BS2_TRACE'
#define BS2_TRACE(val, var) {std::ignore = (val);}
                                           ^~~~~
samples/sample_hevc_fei_abr/include/dso/hevc2_struct.h:785:11: note:
      bit-field is declared here
    Bs16u h : 8;
          ^
/usr/include/c++/v1/tuple:1091:38: note: passing argument to parameter here
    const __ignore_t& operator=(_Tp&&) const {return *this;}
                                     ^
samples/sample_hevc_fei_abr/src/dso/hevc2_ssdata.cpp:645:13: error:
      non-const reference cannot bind to bit-field 'ref_idx_l0'
            BS2_TRACE(pu.ref_idx_l0, pu.ref_idx_l0);
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
samples/sample_hevc_fei_abr/include/dso/bs_reader2.h:187:44: note:
      expanded from macro 'BS2_TRACE'
#define BS2_TRACE(val, var) {std::ignore = (val);}
                                           ^~~~~
samples/sample_hevc_fei_abr/include/dso/hevc2_struct.h:792:11: note:
      bit-field is declared here
    Bs16u ref_idx_l0     : 4;
          ^
/usr/include/c++/v1/tuple:1091:38: note: passing argument to parameter here
    const __ignore_t& operator=(_Tp&&) const {return *this;}
                                     ^
samples/sample_hevc_fei_abr/src/dso/hevc2_ssdata.cpp:647:13: error:
      non-const reference cannot bind to bit-field 'ref_idx_l1'
            BS2_TRACE(pu.ref_idx_l1, pu.ref_idx_l1);
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
samples/sample_hevc_fei_abr/include/dso/bs_reader2.h:187:44: note:
      expanded from macro 'BS2_TRACE'
#define BS2_TRACE(val, var) {std::ignore = (val);}
                                           ^~~~~
samples/sample_hevc_fei_abr/include/dso/hevc2_struct.h:793:11: note:
      bit-field is declared here
    Bs16u ref_idx_l1     : 4;
          ^
/usr/include/c++/v1/tuple:1091:38: note: passing argument to parameter here
    const __ignore_t& operator=(_Tp&&) const {return *this;}
                                     ^
samples/sample_hevc_fei_abr/src/dso/hevc2_ssdata.cpp:710:17: error:
      non-const reference cannot bind to bit-field 'log2CbSize'
                BS2_TRACE(cu.log2CbSize, cu.log2CbSize);
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
samples/sample_hevc_fei_abr/include/dso/bs_reader2.h:187:44: note:
      expanded from macro 'BS2_TRACE'
#define BS2_TRACE(val, var) {std::ignore = (val);}
                                           ^~~~~
samples/sample_hevc_fei_abr/include/dso/hevc2_struct.h:832:11: note:
      bit-field is declared here
    Bs32u log2CbSize             : 8;
          ^
/usr/include/c++/v1/tuple:1091:38: note: passing argument to parameter here
    const __ignore_t& operator=(_Tp&&) const {return *this;}
                                     ^
samples/sample_hevc_fei_abr/src/dso/hevc2_ssdata.cpp:1053:5: error:
      non-const reference cannot bind to bit-field 'w'
    BS2_TRACE(pu.w, pu.w);
    ^~~~~~~~~~~~~~~~~~~~~
samples/sample_hevc_fei_abr/include/dso/bs_reader2.h:187:44: note:
      expanded from macro 'BS2_TRACE'
#define BS2_TRACE(val, var) {std::ignore = (val);}
                                           ^~~~~
samples/sample_hevc_fei_abr/include/dso/hevc2_struct.h:784:11: note:
      bit-field is declared here
    Bs16u w : 8;
          ^
/usr/include/c++/v1/tuple:1091:38: note: passing argument to parameter here
    const __ignore_t& operator=(_Tp&&) const {return *this;}
                                     ^
samples/sample_hevc_fei_abr/src/dso/hevc2_ssdata.cpp:1054:5: error:
      non-const reference cannot bind to bit-field 'h'
    BS2_TRACE(pu.h, pu.h);
    ^~~~~~~~~~~~~~~~~~~~~
samples/sample_hevc_fei_abr/include/dso/bs_reader2.h:187:44: note:
      expanded from macro 'BS2_TRACE'
#define BS2_TRACE(val, var) {std::ignore = (val);}
                                           ^~~~~
samples/sample_hevc_fei_abr/include/dso/hevc2_struct.h:785:11: note:
      bit-field is declared here
    Bs16u h : 8;
          ^
/usr/include/c++/v1/tuple:1091:38: note: passing argument to parameter here
    const __ignore_t& operator=(_Tp&&) const {return *this;}
                                     ^
samples/sample_hevc_fei_abr/src/dso/hevc2_ssdata.cpp:1066:9: error:
      non-const reference cannot bind to bit-field 'inter_pred_idc'
        BS2_TRACE(pu.inter_pred_idc, pu.inter_pred_idc);
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
samples/sample_hevc_fei_abr/include/dso/bs_reader2.h:187:44: note:
      expanded from macro 'BS2_TRACE'
#define BS2_TRACE(val, var) {std::ignore = (val);}
                                           ^~~~~
samples/sample_hevc_fei_abr/include/dso/hevc2_struct.h:789:11: note:
      bit-field is declared here
    Bs16u inter_pred_idc : 2;
          ^
/usr/include/c++/v1/tuple:1091:38: note: passing argument to parameter here
    const __ignore_t& operator=(_Tp&&) const {return *this;}
                                     ^
samples/sample_hevc_fei_abr/src/dso/hevc2_ssdata.cpp:1071:13: error:
      non-const reference cannot bind to bit-field 'ref_idx_l0'
            BS2_TRACE(pu.ref_idx_l0, pu.ref_idx_l0);
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
samples/sample_hevc_fei_abr/include/dso/bs_reader2.h:187:44: note:
      expanded from macro 'BS2_TRACE'
#define BS2_TRACE(val, var) {std::ignore = (val);}
                                           ^~~~~
samples/sample_hevc_fei_abr/include/dso/hevc2_struct.h:792:11: note:
      bit-field is declared here
    Bs16u ref_idx_l0     : 4;
          ^
/usr/include/c++/v1/tuple:1091:38: note: passing argument to parameter here
    const __ignore_t& operator=(_Tp&&) const {return *this;}
                                     ^
samples/sample_hevc_fei_abr/src/dso/hevc2_ssdata.cpp:1073:13: error:
      non-const reference cannot bind to bit-field 'ref_idx_l1'
            BS2_TRACE(pu.ref_idx_l1, pu.ref_idx_l1);
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
samples/sample_hevc_fei_abr/include/dso/bs_reader2.h:187:44: note:
      expanded from macro 'BS2_TRACE'
#define BS2_TRACE(val, var) {std::ignore = (val);}
                                           ^~~~~
samples/sample_hevc_fei_abr/include/dso/hevc2_struct.h:793:11: note:
      bit-field is declared here
    Bs16u ref_idx_l1     : 4;
          ^
/usr/include/c++/v1/tuple:1091:38: note: passing argument to parameter here
    const __ignore_t& operator=(_Tp&&) const {return *this;}
                                     ^
samples/sample_hevc_fei_abr/src/dso/hevc2_ssdata.cpp:1294:5: error:
      non-const reference cannot bind to bit-field 'log2TrafoSize'
    BS2_TRACE(tu.log2TrafoSize, tu.log2TrafoSize);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
samples/sample_hevc_fei_abr/include/dso/bs_reader2.h:187:44: note:
      expanded from macro 'BS2_TRACE'
#define BS2_TRACE(val, var) {std::ignore = (val);}
                                           ^~~~~
samples/sample_hevc_fei_abr/include/dso/hevc2_struct.h:805:11: note:
      bit-field is declared here
    Bs32u log2TrafoSize       : 8;
          ^
/usr/include/c++/v1/tuple:1091:38: note: passing argument to parameter here
    const __ignore_t& operator=(_Tp&&) const {return *this;}
                                     ^
13 errors generated.
```

GCC 9 reports:
```c++
In file included from samples/sample_hevc_fei_abr/include/dso/hevc2_parser.h:23,
                 from samples/sample_hevc_fei_abr/src/dso/hevc2_ssdata.cpp:21:
samples/sample_hevc_fei_abr/src/dso/hevc2_ssdata.cpp: In member function 'void BS_HEVC2::SDParser::parseCU(BS_HEVC2::CU&)':
samples/sample_hevc_fei_abr/src/dso/hevc2_ssdata.cpp:561:18: error: cannot bind bitfield 'cu.BS_HEVC2::CU::log2CbSize' to 'unsigned int&'
  561 |     BS2_TRACE(cu.log2CbSize, cu.log2CbSize);
samples/sample_hevc_fei_abr/include/dso/bs_reader2.h:187:45: note: in definition of macro 'BS2_TRACE'
  187 | #define BS2_TRACE(val, var) {std::ignore = (val);}
      |                                             ^~~
samples/sample_hevc_fei_abr/src/dso/hevc2_ssdata.cpp:635:22: error: cannot bind bitfield 'pu.BS_HEVC2::PU::inter_pred_idc' to 'short unsigned int&'
  635 |         BS2_TRACE(pu.inter_pred_idc, pu.inter_pred_idc);
samples/sample_hevc_fei_abr/include/dso/bs_reader2.h:187:45: note: in definition of macro 'BS2_TRACE'
  187 | #define BS2_TRACE(val, var) {std::ignore = (val);}
      |                                             ^~~
samples/sample_hevc_fei_abr/src/dso/hevc2_ssdata.cpp:641:22: error: cannot bind bitfield 'pu.BS_HEVC2::PU::w' to 'short unsigned int&'
  641 |         BS2_TRACE(pu.w, pu.w);
samples/sample_hevc_fei_abr/include/dso/bs_reader2.h:187:45: note: in definition of macro 'BS2_TRACE'
  187 | #define BS2_TRACE(val, var) {std::ignore = (val);}
      |                                             ^~~
samples/sample_hevc_fei_abr/src/dso/hevc2_ssdata.cpp:642:22: error: cannot bind bitfield 'pu.BS_HEVC2::PU::h' to 'short unsigned int&'
  642 |         BS2_TRACE(pu.h, pu.h);
samples/sample_hevc_fei_abr/include/dso/bs_reader2.h:187:45: note: in definition of macro 'BS2_TRACE'
  187 | #define BS2_TRACE(val, var) {std::ignore = (val);}
      |                                             ^~~
samples/sample_hevc_fei_abr/src/dso/hevc2_ssdata.cpp:645:26: error: cannot bind bitfield 'pu.BS_HEVC2::PU::ref_idx_l0' to 'short unsigned int&'
  645 |             BS2_TRACE(pu.ref_idx_l0, pu.ref_idx_l0);
samples/sample_hevc_fei_abr/include/dso/bs_reader2.h:187:45: note: in definition of macro 'BS2_TRACE'
  187 | #define BS2_TRACE(val, var) {std::ignore = (val);}
      |                                             ^~~
samples/sample_hevc_fei_abr/src/dso/hevc2_ssdata.cpp:647:26: error: cannot bind bitfield 'pu.BS_HEVC2::PU::ref_idx_l1' to 'short unsigned int&'
  647 |             BS2_TRACE(pu.ref_idx_l1, pu.ref_idx_l1);
samples/sample_hevc_fei_abr/include/dso/bs_reader2.h:187:45: note: in definition of macro 'BS2_TRACE'
  187 | #define BS2_TRACE(val, var) {std::ignore = (val);}
      |                                             ^~~
samples/sample_hevc_fei_abr/src/dso/hevc2_ssdata.cpp:710:30: error: cannot bind bitfield 'cu.BS_HEVC2::CU::log2CbSize' to 'unsigned int&'
  710 |                 BS2_TRACE(cu.log2CbSize, cu.log2CbSize);
samples/sample_hevc_fei_abr/include/dso/bs_reader2.h:187:45: note: in definition of macro 'BS2_TRACE'
  187 | #define BS2_TRACE(val, var) {std::ignore = (val);}
      |                                             ^~~
samples/sample_hevc_fei_abr/src/dso/hevc2_ssdata.cpp: In member function 'void BS_HEVC2::SDParser::parsePU(BS_HEVC2::CU&, BS_HEVC2::PU&, Bs16u, Bs16u)':
samples/sample_hevc_fei_abr/src/dso/hevc2_ssdata.cpp:1053:18: error: cannot bind bitfield 'pu.BS_HEVC2::PU::w' to 'short unsigned int&'
 1053 |     BS2_TRACE(pu.w, pu.w);
samples/sample_hevc_fei_abr/include/dso/bs_reader2.h:187:45: note: in definition of macro 'BS2_TRACE'
  187 | #define BS2_TRACE(val, var) {std::ignore = (val);}
      |                                             ^~~
samples/sample_hevc_fei_abr/src/dso/hevc2_ssdata.cpp:1054:18: error: cannot bind bitfield 'pu.BS_HEVC2::PU::h' to 'short unsigned int&'
 1054 |     BS2_TRACE(pu.h, pu.h);
samples/sample_hevc_fei_abr/include/dso/bs_reader2.h:187:45: note: in definition of macro 'BS2_TRACE'
  187 | #define BS2_TRACE(val, var) {std::ignore = (val);}
      |                                             ^~~
samples/sample_hevc_fei_abr/src/dso/hevc2_ssdata.cpp:1066:22: error: cannot bind bitfield 'pu.BS_HEVC2::PU::inter_pred_idc' to 'short unsigned int&'
 1066 |         BS2_TRACE(pu.inter_pred_idc, pu.inter_pred_idc);
samples/sample_hevc_fei_abr/include/dso/bs_reader2.h:187:45: note: in definition of macro 'BS2_TRACE'
  187 | #define BS2_TRACE(val, var) {std::ignore = (val);}
      |                                             ^~~
samples/sample_hevc_fei_abr/src/dso/hevc2_ssdata.cpp:1071:26: error: cannot bind bitfield 'pu.BS_HEVC2::PU::ref_idx_l0' to 'short unsigned int&'
 1071 |             BS2_TRACE(pu.ref_idx_l0, pu.ref_idx_l0);
samples/sample_hevc_fei_abr/include/dso/bs_reader2.h:187:45: note: in definition of macro 'BS2_TRACE'
  187 | #define BS2_TRACE(val, var) {std::ignore = (val);}
      |                                             ^~~
samples/sample_hevc_fei_abr/src/dso/hevc2_ssdata.cpp:1073:26: error: cannot bind bitfield 'pu.BS_HEVC2::PU::ref_idx_l1' to 'short unsigned int&'
 1073 |             BS2_TRACE(pu.ref_idx_l1, pu.ref_idx_l1);
samples/sample_hevc_fei_abr/include/dso/bs_reader2.h:187:45: note: in definition of macro 'BS2_TRACE'
  187 | #define BS2_TRACE(val, var) {std::ignore = (val);}
      |                                             ^~~
samples/sample_hevc_fei_abr/src/dso/hevc2_ssdata.cpp: In member function 'void BS_HEVC2::SDParser::parseTU(BS_HEVC2::CU&, BS_HEVC2::TU&, BS_HEVC2::TU&, Bs16u)':
samples/sample_hevc_fei_abr/src/dso/hevc2_ssdata.cpp:1294:18: error: cannot bind bitfield 'tu.BS_HEVC2::TU::log2TrafoSize' to 'unsigned int&'
 1294 |     BS2_TRACE(tu.log2TrafoSize, tu.log2TrafoSize);
samples/sample_hevc_fei_abr/include/dso/bs_reader2.h:187:45: note: in definition of macro 'BS2_TRACE'
  187 | #define BS2_TRACE(val, var) {std::ignore = (val);}
      |                                             ^~~
```

Also affects `-DBUILD_TOOLS=on`.

Clang 9 reports:
```c++
tools/bs_parser_hevc/src/hevc2_ssdata.cpp:561:5: error: non-const reference cannot bind to bit-field 'log2CbSize'
    BS2_TRACE(cu.log2CbSize, cu.log2CbSize);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
tools/bs_parser_hevc/include/bs_reader2.h:187:44: note: expanded from macro 'BS2_TRACE'
#define BS2_TRACE(val, var) {std::ignore = (val);}
                                           ^~~~~
tools/bs_parser_hevc/include/hevc2_struct.h:832:11: note: bit-field is declared here
    Bs32u log2CbSize             : 8;
          ^
/usr/include/c++/v1/tuple:1091:38: note: passing argument to parameter here
    const __ignore_t& operator=(_Tp&&) const {return *this;}
                                     ^
tools/bs_parser_hevc/src/hevc2_ssdata.cpp:635:9: error: non-const reference cannot bind to bit-field 'inter_pred_idc'
        BS2_TRACE(pu.inter_pred_idc, pu.inter_pred_idc);
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
tools/bs_parser_hevc/include/bs_reader2.h:187:44: note: expanded from macro 'BS2_TRACE'
#define BS2_TRACE(val, var) {std::ignore = (val);}
                                           ^~~~~
tools/bs_parser_hevc/include/hevc2_struct.h:789:11: note: bit-field is declared here
    Bs16u inter_pred_idc : 2;
          ^
/usr/include/c++/v1/tuple:1091:38: note: passing argument to parameter here
    const __ignore_t& operator=(_Tp&&) const {return *this;}
                                     ^
tools/bs_parser_hevc/src/hevc2_ssdata.cpp:641:9: error: non-const reference cannot bind to bit-field 'w'
        BS2_TRACE(pu.w, pu.w);
        ^~~~~~~~~~~~~~~~~~~~~
tools/bs_parser_hevc/include/bs_reader2.h:187:44: note: expanded from macro 'BS2_TRACE'
#define BS2_TRACE(val, var) {std::ignore = (val);}
                                           ^~~~~
tools/bs_parser_hevc/include/hevc2_struct.h:784:11: note: bit-field is declared here
    Bs16u w : 8;
          ^
/usr/include/c++/v1/tuple:1091:38: note: passing argument to parameter here
    const __ignore_t& operator=(_Tp&&) const {return *this;}
                                     ^
tools/bs_parser_hevc/src/hevc2_ssdata.cpp:642:9: error: non-const reference cannot bind to bit-field 'h'
        BS2_TRACE(pu.h, pu.h);
        ^~~~~~~~~~~~~~~~~~~~~
tools/bs_parser_hevc/include/bs_reader2.h:187:44: note: expanded from macro 'BS2_TRACE'
#define BS2_TRACE(val, var) {std::ignore = (val);}
                                           ^~~~~
tools/bs_parser_hevc/include/hevc2_struct.h:785:11: note: bit-field is declared here
    Bs16u h : 8;
          ^
/usr/include/c++/v1/tuple:1091:38: note: passing argument to parameter here
    const __ignore_t& operator=(_Tp&&) const {return *this;}
                                     ^
tools/bs_parser_hevc/src/hevc2_ssdata.cpp:645:13: error: non-const reference cannot bind to bit-field 'ref_idx_l0'
            BS2_TRACE(pu.ref_idx_l0, pu.ref_idx_l0);
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
tools/bs_parser_hevc/include/bs_reader2.h:187:44: note: expanded from macro 'BS2_TRACE'
#define BS2_TRACE(val, var) {std::ignore = (val);}
                                           ^~~~~
tools/bs_parser_hevc/include/hevc2_struct.h:792:11: note: bit-field is declared here
    Bs16u ref_idx_l0     : 4;
          ^
/usr/include/c++/v1/tuple:1091:38: note: passing argument to parameter here
    const __ignore_t& operator=(_Tp&&) const {return *this;}
                                     ^
tools/bs_parser_hevc/src/hevc2_ssdata.cpp:647:13: error: non-const reference cannot bind to bit-field 'ref_idx_l1'
            BS2_TRACE(pu.ref_idx_l1, pu.ref_idx_l1);
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
tools/bs_parser_hevc/include/bs_reader2.h:187:44: note: expanded from macro 'BS2_TRACE'
#define BS2_TRACE(val, var) {std::ignore = (val);}
                                           ^~~~~
tools/bs_parser_hevc/include/hevc2_struct.h:793:11: note: bit-field is declared here
    Bs16u ref_idx_l1     : 4;
          ^
/usr/include/c++/v1/tuple:1091:38: note: passing argument to parameter here
    const __ignore_t& operator=(_Tp&&) const {return *this;}
                                     ^
tools/bs_parser_hevc/src/hevc2_ssdata.cpp:710:17: error: non-const reference cannot bind to bit-field 'log2CbSize'
                BS2_TRACE(cu.log2CbSize, cu.log2CbSize);
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
tools/bs_parser_hevc/include/bs_reader2.h:187:44: note: expanded from macro 'BS2_TRACE'
#define BS2_TRACE(val, var) {std::ignore = (val);}
                                           ^~~~~
tools/bs_parser_hevc/include/hevc2_struct.h:832:11: note: bit-field is declared here
    Bs32u log2CbSize             : 8;
          ^
/usr/include/c++/v1/tuple:1091:38: note: passing argument to parameter here
    const __ignore_t& operator=(_Tp&&) const {return *this;}
                                     ^
tools/bs_parser_hevc/src/hevc2_ssdata.cpp:1053:5: error: non-const reference cannot bind to bit-field 'w'
    BS2_TRACE(pu.w, pu.w);
    ^~~~~~~~~~~~~~~~~~~~~
tools/bs_parser_hevc/include/bs_reader2.h:187:44: note: expanded from macro 'BS2_TRACE'
#define BS2_TRACE(val, var) {std::ignore = (val);}
                                           ^~~~~
tools/bs_parser_hevc/include/hevc2_struct.h:784:11: note: bit-field is declared here
    Bs16u w : 8;
          ^
/usr/include/c++/v1/tuple:1091:38: note: passing argument to parameter here
    const __ignore_t& operator=(_Tp&&) const {return *this;}
                                     ^
tools/bs_parser_hevc/src/hevc2_ssdata.cpp:1054:5: error: non-const reference cannot bind to bit-field 'h'
    BS2_TRACE(pu.h, pu.h);
    ^~~~~~~~~~~~~~~~~~~~~
tools/bs_parser_hevc/include/bs_reader2.h:187:44: note: expanded from macro 'BS2_TRACE'
#define BS2_TRACE(val, var) {std::ignore = (val);}
                                           ^~~~~
tools/bs_parser_hevc/include/hevc2_struct.h:785:11: note: bit-field is declared here
    Bs16u h : 8;
          ^
/usr/include/c++/v1/tuple:1091:38: note: passing argument to parameter here
    const __ignore_t& operator=(_Tp&&) const {return *this;}
                                     ^
tools/bs_parser_hevc/src/hevc2_ssdata.cpp:1066:9: error: non-const reference cannot bind to bit-field 'inter_pred_idc'
        BS2_TRACE(pu.inter_pred_idc, pu.inter_pred_idc);
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
tools/bs_parser_hevc/include/bs_reader2.h:187:44: note: expanded from macro 'BS2_TRACE'
#define BS2_TRACE(val, var) {std::ignore = (val);}
                                           ^~~~~
tools/bs_parser_hevc/include/hevc2_struct.h:789:11: note: bit-field is declared here
    Bs16u inter_pred_idc : 2;
          ^
/usr/include/c++/v1/tuple:1091:38: note: passing argument to parameter here
    const __ignore_t& operator=(_Tp&&) const {return *this;}
                                     ^
tools/bs_parser_hevc/src/hevc2_ssdata.cpp:1071:13: error: non-const reference cannot bind to bit-field 'ref_idx_l0'
            BS2_TRACE(pu.ref_idx_l0, pu.ref_idx_l0);
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
tools/bs_parser_hevc/include/bs_reader2.h:187:44: note: expanded from macro 'BS2_TRACE'
#define BS2_TRACE(val, var) {std::ignore = (val);}
                                           ^~~~~
tools/bs_parser_hevc/include/hevc2_struct.h:792:11: note: bit-field is declared here
    Bs16u ref_idx_l0     : 4;
          ^
/usr/include/c++/v1/tuple:1091:38: note: passing argument to parameter here
    const __ignore_t& operator=(_Tp&&) const {return *this;}
                                     ^
tools/bs_parser_hevc/src/hevc2_ssdata.cpp:1073:13: error: non-const reference cannot bind to bit-field 'ref_idx_l1'
            BS2_TRACE(pu.ref_idx_l1, pu.ref_idx_l1);
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
tools/bs_parser_hevc/include/bs_reader2.h:187:44: note: expanded from macro 'BS2_TRACE'
#define BS2_TRACE(val, var) {std::ignore = (val);}
                                           ^~~~~
tools/bs_parser_hevc/include/hevc2_struct.h:793:11: note: bit-field is declared here
    Bs16u ref_idx_l1     : 4;
          ^
/usr/include/c++/v1/tuple:1091:38: note: passing argument to parameter here
    const __ignore_t& operator=(_Tp&&) const {return *this;}
                                     ^
tools/bs_parser_hevc/src/hevc2_ssdata.cpp:1294:5: error: non-const reference cannot bind to bit-field 'log2TrafoSize'
    BS2_TRACE(tu.log2TrafoSize, tu.log2TrafoSize);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
tools/bs_parser_hevc/include/bs_reader2.h:187:44: note: expanded from macro 'BS2_TRACE'
#define BS2_TRACE(val, var) {std::ignore = (val);}
                                           ^~~~~
tools/bs_parser_hevc/include/hevc2_struct.h:805:11: note: bit-field is declared here
    Bs32u log2TrafoSize       : 8;
          ^
/usr/include/c++/v1/tuple:1091:38: note: passing argument to parameter here
    const __ignore_t& operator=(_Tp&&) const {return *this;}
                                     ^
13 errors generated.
```

GCC 9 reports:
```c++
In file included from tools/bs_parser_hevc/include/hevc2_parser.h:23,
                 from tools/bs_parser_hevc/src/hevc2_ssdata.cpp:21:
tools/bs_parser_hevc/src/hevc2_ssdata.cpp: In member function 'void BS_HEVC2::SDParser::parseCU(BS_HEVC2::CU&)':
tools/bs_parser_hevc/src/hevc2_ssdata.cpp:561:18: error: cannot bind bitfield 'cu.BS_HEVC2::CU::log2CbSize' to 'unsigned int&'
  561 |     BS2_TRACE(cu.log2CbSize, cu.log2CbSize);
tools/bs_parser_hevc/include/bs_reader2.h:187:45: note: in definition of macro 'BS2_TRACE'
  187 | #define BS2_TRACE(val, var) {std::ignore = (val);}
      |                                             ^~~
tools/bs_parser_hevc/src/hevc2_ssdata.cpp:635:22: error: cannot bind bitfield 'pu.BS_HEVC2::PU::inter_pred_idc' to 'short unsigned int&'
  635 |         BS2_TRACE(pu.inter_pred_idc, pu.inter_pred_idc);
tools/bs_parser_hevc/include/bs_reader2.h:187:45: note: in definition of macro 'BS2_TRACE'
  187 | #define BS2_TRACE(val, var) {std::ignore = (val);}
      |                                             ^~~
tools/bs_parser_hevc/src/hevc2_ssdata.cpp:641:22: error: cannot bind bitfield 'pu.BS_HEVC2::PU::w' to 'short unsigned int&'
  641 |         BS2_TRACE(pu.w, pu.w);
tools/bs_parser_hevc/include/bs_reader2.h:187:45: note: in definition of macro 'BS2_TRACE'
  187 | #define BS2_TRACE(val, var) {std::ignore = (val);}
      |                                             ^~~
tools/bs_parser_hevc/src/hevc2_ssdata.cpp:642:22: error: cannot bind bitfield 'pu.BS_HEVC2::PU::h' to 'short unsigned int&'
  642 |         BS2_TRACE(pu.h, pu.h);
tools/bs_parser_hevc/include/bs_reader2.h:187:45: note: in definition of macro 'BS2_TRACE'
  187 | #define BS2_TRACE(val, var) {std::ignore = (val);}
      |                                             ^~~
tools/bs_parser_hevc/src/hevc2_ssdata.cpp:645:26: error: cannot bind bitfield 'pu.BS_HEVC2::PU::ref_idx_l0' to 'short unsigned int&'
  645 |             BS2_TRACE(pu.ref_idx_l0, pu.ref_idx_l0);
tools/bs_parser_hevc/include/bs_reader2.h:187:45: note: in definition of macro 'BS2_TRACE'
  187 | #define BS2_TRACE(val, var) {std::ignore = (val);}
      |                                             ^~~
tools/bs_parser_hevc/src/hevc2_ssdata.cpp:647:26: error: cannot bind bitfield 'pu.BS_HEVC2::PU::ref_idx_l1' to 'short unsigned int&'
  647 |             BS2_TRACE(pu.ref_idx_l1, pu.ref_idx_l1);
tools/bs_parser_hevc/include/bs_reader2.h:187:45: note: in definition of macro 'BS2_TRACE'
  187 | #define BS2_TRACE(val, var) {std::ignore = (val);}
      |                                             ^~~
tools/bs_parser_hevc/src/hevc2_ssdata.cpp:710:30: error: cannot bind bitfield 'cu.BS_HEVC2::CU::log2CbSize' to 'unsigned int&'
  710 |                 BS2_TRACE(cu.log2CbSize, cu.log2CbSize);
tools/bs_parser_hevc/include/bs_reader2.h:187:45: note: in definition of macro 'BS2_TRACE'
  187 | #define BS2_TRACE(val, var) {std::ignore = (val);}
      |                                             ^~~
tools/bs_parser_hevc/src/hevc2_ssdata.cpp: In member function 'void BS_HEVC2::SDParser::parsePU(BS_HEVC2::CU&, BS_HEVC2::PU&, Bs16u, Bs16u)':
tools/bs_parser_hevc/src/hevc2_ssdata.cpp:1053:18: error: cannot bind bitfield 'pu.BS_HEVC2::PU::w' to 'short unsigned int&'
 1053 |     BS2_TRACE(pu.w, pu.w);
tools/bs_parser_hevc/include/bs_reader2.h:187:45: note: in definition of macro 'BS2_TRACE'
  187 | #define BS2_TRACE(val, var) {std::ignore = (val);}
      |                                             ^~~
tools/bs_parser_hevc/src/hevc2_ssdata.cpp:1054:18: error: cannot bind bitfield 'pu.BS_HEVC2::PU::h' to 'short unsigned int&'
 1054 |     BS2_TRACE(pu.h, pu.h);
tools/bs_parser_hevc/include/bs_reader2.h:187:45: note: in definition of macro 'BS2_TRACE'
  187 | #define BS2_TRACE(val, var) {std::ignore = (val);}
      |                                             ^~~
tools/bs_parser_hevc/src/hevc2_ssdata.cpp:1066:22: error: cannot bind bitfield 'pu.BS_HEVC2::PU::inter_pred_idc' to 'short unsigned int&'
 1066 |         BS2_TRACE(pu.inter_pred_idc, pu.inter_pred_idc);
tools/bs_parser_hevc/include/bs_reader2.h:187:45: note: in definition of macro 'BS2_TRACE'
  187 | #define BS2_TRACE(val, var) {std::ignore = (val);}
      |                                             ^~~
tools/bs_parser_hevc/src/hevc2_ssdata.cpp:1071:26: error: cannot bind bitfield 'pu.BS_HEVC2::PU::ref_idx_l0' to 'short unsigned int&'
 1071 |             BS2_TRACE(pu.ref_idx_l0, pu.ref_idx_l0);
tools/bs_parser_hevc/include/bs_reader2.h:187:45: note: in definition of macro 'BS2_TRACE'
  187 | #define BS2_TRACE(val, var) {std::ignore = (val);}
      |                                             ^~~
tools/bs_parser_hevc/src/hevc2_ssdata.cpp:1073:26: error: cannot bind bitfield 'pu.BS_HEVC2::PU::ref_idx_l1' to 'short unsigned int&'
 1073 |             BS2_TRACE(pu.ref_idx_l1, pu.ref_idx_l1);
tools/bs_parser_hevc/include/bs_reader2.h:187:45: note: in definition of macro 'BS2_TRACE'
  187 | #define BS2_TRACE(val, var) {std::ignore = (val);}
      |                                             ^~~
tools/bs_parser_hevc/src/hevc2_ssdata.cpp: In member function 'void BS_HEVC2::SDParser::parseTU(BS_HEVC2::CU&, BS_HEVC2::TU&, BS_HEVC2::TU&, Bs16u)':
tools/bs_parser_hevc/src/hevc2_ssdata.cpp:1294:18: error: cannot bind bitfield 'tu.BS_HEVC2::TU::log2TrafoSize' to 'unsigned int&'
 1294 |     BS2_TRACE(tu.log2TrafoSize, tu.log2TrafoSize);
tools/bs_parser_hevc/include/bs_reader2.h:187:45: note: in definition of macro 'BS2_TRACE'
  187 | #define BS2_TRACE(val, var) {std::ignore = (val);}
      |                                             ^~~
```

--- a/samples/sample_hevc_fei_abr/include/dso/bs_reader2.h
+++ b/samples/sample_hevc_fei_abr/include/dso/bs_reader2.h
@@ -1,4 +1,4 @@
-// Copyright (c) 2018-2019 Intel Corporation
+// Copyright (c) 2018-2020 Intel Corporation
 //
 // Permission is hereby granted, free of charge, to any person obtaining a copy
 // of this software and associated documentation files (the "Software"), to deal
@@ -184,16 +184,16 @@ public:
 
 #define BS2_TRO
 #define BS2_SET(val, var) { var = (val); }
-#define BS2_TRACE(val, var) {std::ignore = (val);}
-#define BS2_TRACE_STR(str) {std::ignore = (str);}
+#define BS2_TRACE(val, var) {(void)(val);}
+#define BS2_TRACE_STR(str) {(void)(str);}
 #define BS2_SET_ARR_F(val, var, sz, split, format) \
     { for (Bs32u _i = 0; _i < (Bs32u)(sz); _i++) (var)[_i] = (val); }
 #define BS2_TRACE_ARR_VF(val, var, sz, split, format) \
-    { for (Bs32u _i = 0; _i < (Bs32u)(sz); _i++) std::ignore = (val); }
-#define BS2_TRACE_MDARR(type, arr, dim, split, split2, format) {std::ignore = (arr);}
+    { for (Bs32u _i = 0; _i < (Bs32u)(sz); _i++) (void)(val); }
+#define BS2_TRACE_MDARR(type, arr, dim, split, split2, format) {(void)(arr);}
 #define BS2_SET_ARR_M(val, var, sz, split, format, map) BS2_SET_ARR_F(val, var, sz, split, format)
 #define BS2_SETM(val, var, map) BS2_SET(val, var)
-#define BS2_TRACE_BIN(pval, off, sz, var) { std::ignore = (pval); }
+#define BS2_TRACE_BIN(pval, off, sz, var) { (void)(pval); }
 
 #endif
 

--- a/tools/bs_parser_hevc/include/bs_reader2.h
+++ b/tools/bs_parser_hevc/include/bs_reader2.h
@@ -1,4 +1,4 @@
-// Copyright (c) 2018-2019 Intel Corporation
+// Copyright (c) 2018-2020 Intel Corporation
 //
 // Permission is hereby granted, free of charge, to any person obtaining a copy
 // of this software and associated documentation files (the "Software"), to deal
@@ -184,16 +184,16 @@ public:
 
 #define BS2_TRO
 #define BS2_SET(val, var) { var = (val); }
-#define BS2_TRACE(val, var) {std::ignore = (val);}
-#define BS2_TRACE_STR(str) {std::ignore = (str);}
+#define BS2_TRACE(val, var) {(void)(val);}
+#define BS2_TRACE_STR(str) {(void)(str);}
 #define BS2_SET_ARR_F(val, var, sz, split, format) \
     { for (Bs32u _i = 0; _i < (Bs32u)(sz); _i++) (var)[_i] = (val); }
 #define BS2_TRACE_ARR_VF(val, var, sz, split, format) \
-    { for (Bs32u _i = 0; _i < (Bs32u)(sz); _i++) std::ignore = (val); }
+    { for (Bs32u _i = 0; _i < (Bs32u)(sz); _i++) (void)(val); }
 #define BS2_TRACE_MDARR(type, arr, dim, split, split2, format) {std::ignore = (arr);}
 #define BS2_SET_ARR_M(val, var, sz, split, format, map) BS2_SET_ARR_F(val, var, sz, split, format)
 #define BS2_SETM(val, var, map) BS2_SET(val, var)
-#define BS2_TRACE_BIN(pval, off, sz, var) { std::ignore = (pval); }
+#define BS2_TRACE_BIN(pval, off, sz, var) { (void)(pval); }
 
 #endif
 

